awslabs/performance-dashboard-on-aws
# GTT-926 Implement SSO with SAML Identity Provider

## Description

**This PR is to enable a code review, and will not be merged into mainline**.  After reviewing, the changes will remain in a feature branch until the feature has been approved by Security.  After that, it will be merged into mainline

GTT-926, GTT-871

* Replaced withAuthenticator with withSAMLAuthenticator to expose "enterprise sign-in" button which calls federatedSignIn() to Identity Provider (IdP)
*  Added properties to EnvConfig related to federated login
*  Added properties to amplify-config required by Amplify federatedSignIn
*  Updated user-hooks to look for ["custom:groups"] in addition to ["custom:roles"] in Id token when doing federated authentication
* Updated auth.ts to look for ["custom:groups"] in Id token in addition to ["custom:roles"]
*  Modified logout to do single sign-out

## Testing

Tested in my AWS [environment](https://trietlu.badger.wwps.aws.dev/admin)
* login and logout as federated users with Admin and Editor roles (separately)
* login and logout as Cognito user
* tested single sign-out

Will be adding automated API test in another PR

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
